### PR TITLE
Support dark mode

### PR DESCRIPTION
**Pull request by alishari**
I have implemented dark mode support using material colors #450

### CODE REVIEW

**Review by hbb20 — CHANGES_REQUESTED** [review]
Thank you for the PR. I am sure this will be helpful to many devs. Although, we should correct the colors of demo app itself. Like following:
image

File: app/src/main/res/layout/fragment_full_number.xml [hunk 1]
```diff
@@ -75,7 +75,7 @@
                     android:layout_marginBottom="16dp"
                     android:layout_marginLeft="16dp"
                     android:layout_marginRight="16dp"
-                    android:background="#e5e5e5"
+                    android:background="?attr/colorSurface"
```
Comment (darcishED) [inline app/src/main/res/layout/fragment_full_number.xml:78]: [minor] could be shortened to ?/colroSurface .. it doesn't need to be explicitly defined

Comment (darcishED) [inline app/src/main/res/layout/fragment_full_number.xml:78]: [minor] could be shortened to ?/colroSurface .. it doesn't need to be explicitly defined

File: app/src/main/res/layout/fragment_full_number.xml [hunk 2]
```diff
@@ -140,7 +140,7 @@
                     android:layout_height="wrap_content"
                     android:layout_marginLeft="16dp"
                     android:layout_marginRight="16dp"
-                    android:background="#e5e5e5"
+                    android:background="?attr/colorSurface"
```
Comment (darcishED) [inline app/src/main/res/layout/fragment_full_number.xml:143]: same here

File: app/src/main/res/layout/fragment_introduction.xml [hunk 3]
```diff
@@ -88,7 +88,7 @@
                     android:layout_marginLeft="16dp"
                     android:layout_marginRight="16dp"
                     android:layout_marginBottom="16dp"
-                    android:background="#e5e5e5"
+                    android:background="?attr/colorSurface"
```
Comment (darcishED) [inline app/src/main/res/layout/fragment_introduction.xml:91]: same here

File: app/src/main/res/layout/fragment_introduction.xml [hunk 4]
```diff
@@ -49,7 +49,7 @@
...
```
Comment (darcishED) [inline app/src/main/res/layout/fragment_introduction.xml:52]: same here

File: app/build.gradle [hunk 5]
```diff
@@ -34,4 +34,5 @@ dependencies {
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
 	implementation 'androidx.cardview:cardview:1.0.0'
 	testImplementation 'junit:junit:4.13.2'
+    implementation 'com.google.android.material:material:1.4.0'
```
Comment (darcishED) [inline app/build.gradle:37]: update version

File: ccp/src/main/res/layout/layout_recycler_country_tile.xml [hunk 6]
```diff
@@ -48,12 +47,11 @@
         android:layout_marginEnd="8dp"
         android:layout_marginRight="8dp"
         android:gravity="center_vertical"
-        android:paddingEnd="10dp"
+        android:paddingStart="10dp"
         android:paddingLeft="10dp"
```
Comment (darcishED) [inline ccp/src/main/res/layout/layout_recycler_country_tile.xml:51]: Why keep this if there is paddindStart?

**Review by darcishED — COMMENTED** [review]
[EOF} Add an empty line at the end of all files

**Pull request closed without merging**